### ISSUE TITLE
Seastar.unit.alloc fails

### ISSUE THREAD

**Issue by tchaikov** [scylladb/seastar#1224]
```
      Start  2: Seastar.unit.alloc
 2/73 Test  #2: Seastar.unit.alloc ............................***Failed    0.49 sec
[0/1] cd /home/circleci/project/build/release/tests/unit && /home/circleci/project/build/release/tests/unit/alloc_test -- -c 2
Running 9 test cases...
WARN  2022-09-28 13:35:34,866 [shard 0] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
INFO  2022-09-28 13:35:34,867 [shard 0] seastar - Created fair group io-queue-0, capacity rate 2147483:2147483, limit 12582912, rate 16777216 (factor 1), threshold 2000
INFO  2022-09-28 13:35:34,867 [shard 0] seastar - IO queue uses 0.75ms latency goal for device 0
INFO  2022-09-28 13:35:34,867 [shard 0] seastar - Created io group dev(0), length limit 4194304:4194304, rate 2147483647:2147483647
INFO  2022-09-28 13:35:34,868 [shard 0] seastar - Created io queue dev(0) capacities: 512:2000:2000 1024:3000:3000 2048:5000:5000 4096:9000:9000 8192:17000:17000 16384:33000:33000 32768:65000:65000 65536:129000:129000 131072:257000:257000
WARN  2022-09-28 13:35:34,892 [shard 1] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
random-seed=1775285472
/home/circleci/project/tests/unit/alloc_test.cc(209): fatal error: in "test_large_allocation_warning_off_by_one": critical check memory::stats().large_allocations() == old_large_allocs_count+1 has failed

*** 1 failure is detected in the test module "/home/circleci/project/tests/unit/alloc_test.cc"
```

not able to reproduce locally with release build.

**Comment by raphaelsc** [comment]
Looking

**Comment by tchaikov** [comment]
should have been fixed by 92fdfec585ad41a3f36159eacf9a784a3ee960db
